src/Core/Scheduler/Providers/GpxUtils.js

```diff
@@ -16,78 +20,173 @@ function _gGpxToWTrackPointsArray(gpxXML) {
     return gpxXML.getElementsByTagName('trkpt');
 }
 
-function _gpxPtToCartesian(pt) {
+function _gpxPtToCartesian(pt, crs) {
     var longitude = Number(pt.attributes.lon.nodeValue);
     var latitude = Number(pt.attributes.lat.nodeValue);
-    var elevation = Number(pt.getElementsByTagName('ele')[0].childNodes[0].nodeValue);
+    // TODO: get elevation with terrain
+    const elem = pt.getElementsByTagName('ele')[0];
+    const elevation = elem ? Number(elem.childNodes[0].nodeValue) : 0;
 
-    return new Coordinates('EPSG:4326', longitude, latitude, elevation).as('EPSG:4978').xyz();
+    return new Coordinates('EPSG:4326', longitude, latitude, elevation).as(crs).xyz();
 }
 
-function _gpxToWayPointsMesh(gpxXML) {
+const geometryPoint = new THREE.BoxGeometry(1, 1, 80);
+const materialPoint = new THREE.MeshBasicMaterial({ color: 0xffffff });
+const point = new THREE.Vector3();
+
+function getDistance(object, camera) {
+    point.setFromMatrixPosition(object.matrixWorld).add(object.geometry.boundingSphere.center);
+    return camera.position.distanceTo(point);
```

Comment:
This ignores camera's transformation as well (but this isn't the only place in itowns that assumes that the camera has no parent)